wallet-new/src/Cardano/Wallet/Server/LegacyPlugins.hs

```diff
@@ -0,0 +1,226 @@
+    , notifierPlugin
+    ) where
+
+import           Universum
+
+import           Cardano.Wallet.API as API
+import           Cardano.Wallet.API.V1.Headers (applicationJson)
+import qualified Cardano.Wallet.API.V1.Types as V1
+import qualified Cardano.Wallet.LegacyServer as LegacyServer
+import           Cardano.Wallet.Server.CLI (RunMode, WalletBackendParams (..),
+                     isDebugMode, walletAcidInterval, walletDbOptions)
+import qualified Pos.Wallet.Web.Error.Types as V0
+
+import           Control.Exception (fromException)
+import           Data.Aeson
+import           Formatting (build, sformat, (%))
+import           Network.HTTP.Types.Status (badRequest400)
+import           Network.Wai (Application, Middleware, Response, responseLBS)
+import           Network.Wai.Handler.Warp (defaultSettings,
+                     setOnExceptionResponse)
+import           Network.Wai.Middleware.Cors (cors, corsMethods,
+                     corsRequestHeaders, simpleCorsResourcePolicy,
+                     simpleMethods)
+import           Ntp.Client (NtpStatus)
+import           Pos.Chain.Txp (TxpConfiguration)
+import           Pos.Infra.Diffusion.Types (Diffusion (..))
+import           Pos.Wallet.Web (cleanupAcidStatePeriodically)
+import           Pos.Wallet.Web.Pending.Worker (startPendingTxsResubmitter)
+import qualified Pos.Wallet.Web.Server.Runner as V0
+import           Pos.Wallet.Web.Sockets (getWalletWebSockets,
+                     upgradeApplicationWS)
+import qualified Servant
+
+import           Pos.Context (HasNodeContext)
+import           Pos.Core as Core (BlockCount, Config)
+import           Pos.Util (lensOf)
+import           Pos.Util.Wlog (logInfo, modifyLoggerName, usingLoggerName)
+
+import           Cardano.NodeIPC (startNodeJsIPC)
+import           Pos.Configuration (walletProductionApi,
+                     walletTxCreationDisabled)
+import           Pos.Infra.Shutdown.Class (HasShutdownContext (shutdownContext))
+import           Pos.Launcher.Configuration (HasConfigurations)
+import           Pos.Util.CompileInfo (HasCompileInfo)
+import           Pos.Wallet.Web.Mode (WalletWebMode)
+import           Pos.Wallet.Web.Server.Launcher (walletDocumentationImpl,
+                     walletServeImpl)
+import           Pos.Wallet.Web.State (askWalletDB)
+import           Pos.Wallet.Web.Tracking.Sync (processSyncRequest)
+import           Pos.Wallet.Web.Tracking.Types (SyncQueue)
+import           Pos.Web (serveWeb)
+import           Pos.WorkMode (WorkMode)
+
+
+-- A @Plugin@ running in the monad @m@.
+type Plugin m = [Diffusion m -> m ()]
+
+-- | A @Plugin@ to periodically compact & snapshot the acid-state database.
+acidCleanupWorker :: WalletBackendParams
+                  -> Plugin WalletWebMode
+acidCleanupWorker WalletBackendParams{..} = pure $ const $
+    modifyLoggerName (const "acidcleanup") $
+    askWalletDB >>= \db -> cleanupAcidStatePeriodically db (walletAcidInterval walletDbOptions)
+
```

Comment:
won't the new wallet use this acidCleanupWorker? I see it`s moved in Legacy.

Reply:
We will rewrite it as part of CBR-365 😉